src/lib/kdb/kdb5.c

```diff
@@ -2564,3 +2602,28 @@ krb5_db_check_allowed_to_delegate(krb5_context kcontext,
         return KRB5_PLUGIN_OP_NOTSUPP;
     return v->check_allowed_to_delegate(kcontext, client, server, proxy);
 }
+
+#define COMPARE_KEY_DATA(k1, k2) k1.key_data_kvno < k2.key_data_kvno
```

Comment:
A static inline function would probably be better than a macro here.  If you must use a macro, the result should be enclosed in parentheses as should k1 and k2 (see http://www.cs.yale.edu/homes/aspnes/pinewiki/C(2f)Macros.html ).  "compare" is also a somewhat deceptive name for a less-than check; idiomatically C comparison functions behave like strcmp() in their return values.